### ISSUE TITLE
external dependency definition file

### ISSUE THREAD

**Issue by JakeDEvans** [getsentry/onpremise#769]
## Summary

Please consolidate all external dependencies such as docker images (like alpine, busybox, postgres, zookeeper, kafka, clickhouse, nginx, exim) and package distributions (like buster debian) in one place so that this project can be easily installed and maintained behind highly firewalled and internet disconnect systems where package distribution systems like artifactory may be used.

## Motivation

More clearly define external dependencies to make maintaining necessary internal changes manageable.

**Comment by chadwhitacre** [comment]
Seems reasonable. So we would pull all images from a single repository, which would be one that we (Sentry) control by default, but would be easy for you to override to point to your artifactory? Something like that?

**Comment by JakeDEvans** [comment]
Yes! not having to build locally would be preferred, if you can publish all images to a publicly accessible docker registry then it would be easy to override the images with those paths.

**Comment by chadwhitacre** [comment]
Makes sense. Marking as an enhancement, will see what kind of additional demand we get and if/when we can prioritize internally or see an external PR. 😁

Thanks for the suggestion and for using Sentry! :D

**Comment by BYK** [comment]
@chadwhitacre I guess we can build the cron images periodically ourselves?

**Comment by chadwhitacre** [comment]
Sorry ... what are the cron images?

**Comment by JakeDEvans** [comment]
> Sorry ... what are the cron images?

https://github.com/getsentry/onpremise/blob/master/cron/Dockerfile

**Comment by chadwhitacre** [comment]
Gotcha, seeing it now for [snuba](https://github.com/getsentry/onpremise/blob/20.12.1/docker-compose.yml#L185-L194) and [symbolicator](https://github.com/getsentry/onpremise/blob/20.12.1/docker-compose.yml#L185-L194) cleanups (and I remember considering it for refreshing geoip data but deciding on [install-time](https://github.com/getsentry/onpremise/blob/20.12.1/docker-compose.yml#L4-L7) instead). I guess it'd be not just the cron images we'd have to build periodically but also [everything](https://github.com/getsentry/onpremise/blob/20.12.1/docker-compose.yml#L195-L215) that uses [`&sentry_defaults`](https://github.com/getsentry/onpremise/blob/20.12.1/docker-compose.yml#L4-L7), no?

Why do we currently build on install vs. build-once? It has to do with configuration, I think?

**Comment by BYK** [comment]
>Why do we currently build on install vs. build-once? It has to do with configuration, I think?

Not really the configuration at the moment. We had this for Sentry image where we embedded the config files into the locally-built image but we no longer do that. I don't see any good reason to build these images at install time except for the CI and storage costs we'd be paying for building and maintaining these images ourselves.

**Comment by chadwhitacre** [comment]
> CI and storage costs we'd be paying for building and maintaining these images ourselves.

CI we probably end up doing anyway somewhere along the line, and Docker Hub is [free for public images](https://www.docker.com/pricing).
